modules/src/core/ics02_client/client_consensus.rs

```diff
@@ -23,23 +23,30 @@ pub const TENDERMINT_CONSENSUS_STATE_TYPE_URL: &str =
 
 pub const MOCK_CONSENSUS_STATE_TYPE_URL: &str = "/ibc.mock.ConsensusState";
 
-pub trait ConsensusState: Clone + core::fmt::Debug + Send + Sync {
-    type Error;
-
+pub trait ConsensusState: core::fmt::Debug + Send + Sync + AsAny {
     /// Type of client associated with this consensus state (eg. Tendermint)
     fn client_type(&self) -> ClientType;
 
     /// Commitment root of the consensus state, which is used for key-value pair verification.
     fn root(&self) -> &CommitmentRoot;
 
-    /// Wrap into an `AnyConsensusState`
-    fn wrap_any(self) -> AnyConsensusState;
+    fn encode_vec(&self) -> Result<Vec<u8>, Error>;
+}
+
+pub trait AsAny: CoreAny {
```

Comment:
I think we can remove the `CoreAny` trait bound, right?

Reply:
Right, I think we can remove it, but I feel like it's better to keep it because `AsAny` can only be implemented for types that satisfy the `'static` bound (i.e. for which `Any` is implemented).